assignment1/Asiign1_cal.c

```diff
@@ -0,0 +1,131 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <conio.h>
+#include <string.h>
+#include <ctype.h>
+
+#define MAX_SIZE 100
+
+int precedence(char op)
+{
+    if (op == '+' || op == '-')
+        return 1;
+
+    if (op == '*' || op == '/')
+        return 2;
+
+    return 0;
+}
+
+int operation(int a, int b, char op)
+{
+    switch (op)
+    {
+    case '+':
+        return a + b;
+    case '-':
+        return a - b;
+    case '*':
+        return a * b;
+
+    case '/':
+        if (b == 0)
+        {
+
+            printf("Error : Division by zero. \n");
+           exit(EXIT_FAILURE);
+         
+        }
+        return a / b;
+    }
+    return 0;
+}
+
+int evaluateExpression(char *exp)
+{
+    int values[MAX_SIZE], valueTop = -1;
+    char operator[MAX_SIZE], operatorTop = -1;
+
+    for (int i = 0; i < strlen(exp); i++)
+    {
+
+        if (isspace(exp[i]))
+        {
+            continue;
+        }
+
+        if (isdigit(exp[i]))
+        {
+
+            int value = 0;
+
+            while (i < strlen(exp) && isdigit(exp[i]))
```

Comment:
No need to call strlen(exp) function repeatedly; store the value and use it in the required place.